server/src/test/java/io/crate/execution/ddl/tables/RenameColumnTaskTest.java

```diff
@@ -0,0 +1,460 @@
+
+    @Test
+    public void test_rename_simple_columns() throws Exception {
+        // the test is equivalent to : alter table tbl rename column x to y;
+        var e = SQLExecutor.builder(clusterService)
+            .addTable("create table tbl (x int)")
+            .build();
+        DocTableInfo tbl = e.resolveTableInfo("tbl");
+        try (IndexEnv indexEnv = new IndexEnv(
+            THREAD_POOL,
+            tbl,
+            clusterService.state(),
+            Version.CURRENT,
+            createTempDir()
+        )) {
+            var renameColumnTask = new RenameColumnTask(e.nodeCtx, imd -> indexEnv.mapperService());
+            SimpleReference refToRename = new SimpleReference(
+                new ReferenceIdent(tbl.ident(), "x"),
+                RowGranularity.DOC,
+                DataTypes.INTEGER,
+                ColumnPolicy.DYNAMIC,
+                IndexType.PLAIN,
+                true,
+                true,
+                1,
+                1,
+                false,
+                null
+            );
+            var newName = new ColumnIdent("y");
+            var request = new RenameColumnRequest(tbl.ident(), new AnalyzedAlterTableRenameColumn.RenameColumn(refToRename, newName));
+            ClusterState newState = renameColumnTask.execute(clusterService.state(), request);
+            DocTableInfo newTable = new DocTableInfoFactory(e.nodeCtx).create(tbl.ident(), newState.metadata());
+            assertThat(newTable.getReference(refToRename.column())).isNull();
+            assertThat(newTable.getReference(newName)).isReference().hasName(newName.sqlFqn());
+        }
+    }
+
+    @Test
+    public void test_rename_nested_columns() throws Exception {
+        // equivalent to:
+        // alter table tbl rename column o to p;
+        // alter table tbl rename column p['o2'] to p['p2'];
+        // alter table tbl rename column p['p2']['x'] to p['p2']['y'];
+        var e = SQLExecutor.builder(clusterService)
+            .addTable("create table tbl (o object as (o2 object as (x int)))")
+            .build();
+        DocTableInfo tbl = e.resolveTableInfo("tbl");
+        try (IndexEnv indexEnv = new IndexEnv(
+            THREAD_POOL,
+            tbl,
+            clusterService.state(),
+            Version.CURRENT,
+            createTempDir()
+        )) {
+            var renameColumnTask = new RenameColumnTask(e.nodeCtx, imd -> indexEnv.mapperService());
+            SimpleReference refToRename1 = new SimpleReference(
+                new ReferenceIdent(tbl.ident(), new ColumnIdent("o")),
+                RowGranularity.DOC,
+                ObjectType.builder().setInnerType("o2", ObjectType.builder().setInnerType("x", DataTypes.INTEGER).build()).build(),
+                ColumnPolicy.DYNAMIC,
+                IndexType.PLAIN,
+                true,
+                false,
+                1,
+                1,
+                false,
+                null
+            );
```

Comment:
Could simplify the col-to-rename definitions by using `tbl.getReference(new ColumnIdent("o"))`. Would also avoid mistakenly using different properties in the SimpleReference